sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSourceTransformWithStateSuite.scala

```diff
@@ -481,4 +481,94 @@ class StateDataSourceTransformWithStateSuite extends StateStoreMetricsTest
       }
     }
   }
```

Comment:
Can we add a test case for readRegisteredTimers = false to verify exceptions are thrown?

Reply:
Done